dwi_ml/experiment/trainor_abstract.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import os
+
+import time
+
+import numpy as np
+import torch
+
+from dwi_ml.data.dataset.data_list import \
+    (MultiSubjectDataset, LazyMultiSubjectDataset)
+
+# À ARRANGER
+from scil_vital.shared.code.utils.timer import Timer
+from scil_vital.shared.code.utils.experiment_utils import ValueMonitor
+
+
+class TractoMLAbstractLocal:
+    def __init__(self):
+        raise NotImplementedError
+
+    def build_model(self):
+        raise NotImplementedError
+
+    def train(self, **kwargs):
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+    def load_model(self, filepath, **kwargs):
+        raise NotImplementedError
+
+
+class TractoMLAbstractSequences:
+    def __init__(self,
+                 train_database_path: str = None,
+                 valid_database_path: str = None,
+                 name: str = None,
+                 # Concerning the choice of inputs:
+                 nb_degree_angles: int = 128,
+                 add_streamline_noise: bool = False,
+                 streamlines_cut_ratio: float = None, step_size: float = None,
+                 neighborhood_dist_mm: float = None,
+                 nb_neighborhood_axes: int = 6,
+                 add_previous_dir: bool = False,
+                 lazy: bool = False,
+                 # Concerning the memory usage:
+                 batch_size: int = 20000, volumes_per_batch: int = None,
+                 cycles_per_volume_batch: int = 1,
+                 n_epoch: int = 100, seed: int = 1234, patience: int = 20,
+                 use_gpu: bool = True, num_workers: int = 0,
+                 worker_interpolation: bool = False,
+                 cache_manager: bool = False, taskman_managed: bool = False):
+        """
+        Mandatory parameters:
+        ---------------------
+        train_database_path : str
```

Comment:
They have a default value, so it looks like it is not mandatory.

Also, are we specifying the types with the default values or just in the documentation? Having them on both places increases the chances of having inconsistencies.

Reply:
Thanks, removed default value.

Pycharm tells me if the documentation type doesn't fit with the signature type, there shouldn't be problems here I think.

Reply:
Yes, IDEs will warn us about it, but, unfortunately, more frequently than not we tend to ignore warnings because we have more important issues. If we put the type only in one place, we will not be mistaken. Else, we will need automatic tools outside our IDEs to warn us and may be prevent us from merging when these things are detected in a PR.